File: backend/src/Fix.Application/UseCases/Policies/Commands/UpdatePolicyCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Policies.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Policies.Commands;

[RequireRole(RoleCodes.UpdatePolicy)]
public sealed record UpdatePolicyCommand(
    Guid UserId,
    Guid OrganizationId,
    Guid Id,
    string Code,
    string Title,
    string? Description,
    DateOnly ValidFrom,
    DateOnly? ValidTo)
    : ICommand<PolicyDto>, IOrganizationRequest;

File: frontend/bff/src/application/policies/commands/update-policy.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface UpdatePolicyCommand {
  context: RequestContext;
  id: string;
  code: string;
  title: string;
  description: string | null;
  validFrom: string;
  validTo: string | null;
}
